libs/base-ui/contexts/Analytics.tsx

```diff
@@ -10,62 +10,60 @@ export type AnalyticsContextProps = {
 };
 
 export const AnalyticsContext = createContext<AnalyticsContextProps>({
-  logEventWithContext: function () {
-    return undefined;
-  },
+  logEventWithContext: () => undefined,
   fullContext: '',
 });
 
 export function useAnalytics() {
   const context = useContext(AnalyticsContext);
-  if (context === undefined) {
-    throw new Error('useAnalytics must be used within a AnalyticsProvider');
+  if (!context) {
+    throw new Error('useAnalytics must be used within an AnalyticsProvider');
   }
   return context;
 }
 
 type AnalyticsProviderProps = {
   children?: ReactNode;
-  context: string; // TODO: This could be an enum in CCAEventData
+  context: string;
 };
 
 export default function AnalyticsProvider({ children, context }: AnalyticsProviderProps) {
-  const { fullContext: previousContext } = useAnalytics();
-
-  const fullContext = [previousContext, context].filter((c) => !!c).join('_');
```

Comment:
please revert your change. this code creates a snake-cased hierarchy of nested analytics providers